velox/functions/lib/aggregates/DecimalAggregate.h

```diff
@@ -338,4 +338,37 @@ class DecimalAggregate : public exec::Aggregate {
   DecodedVector decodedPartial_;
 };
 
+/* This method is used to correct the sum result calculated using
+ * addWithOverflow. In the addWithOverflow method, when summing two int128_t
+ * with the same sign, it first converts them to uint128_t for calculation. The
+ * returned sum represents the lower 127 bits of the uint128_t sum. The overflow
+ * variable indicates whether an overflow occurred during the calculation. If
+ * overflow = 1, it means there was an upward overflow, and if overflow = -1,
+ * there was a downward overflow. However, since the sum only retains the lower
+ * 127 bits, it may miss one calculation of +(1 << 127) or -(1 << 127).
+ * Therefore, we need to make the following adjustments:
+ * 1. If overflow = 1 && sum < 0, it means that the calculation process missed
+ * one calculation of +(1 << 127), it needs to be corrected to
+ * +(1 << 127) + sum.
+ * 2. If overflow = -1 && sum > 0, it means that the calculation process missed
+ * one calculation of -(1 << 127), it needs to be corrected to
+ * -(1 << 127) + sum.
+ *
+ */
+inline static std::optional<int128_t> computeValidSum(
+    int128_t sum,
+    int64_t overflow) {
+  // Value is valid if the conditions below are true.
+  int128_t validSum = sum;
```

Comment:
the term "valid sum" sounds a bit strange; it suggests that some sums are valid and some are not, but shouldn't all calculations in a database engine be valid?

Maybe remove this variable altogether:

```
if (...) {
   return static_cast<int128_t>(...
}

if (overflow != 0) {
   return std::nullopt;
}

return sum;
```